### PR TITLE
HW 36 Yury

### PR DESCRIPTION
**Pull request by Yury-Odinson**
It's so difficult. I don't fully understand reduce(). 
on the web are many examples with sum of numbers (it's easy), but when I work with array which is in objects (which are also in an array :) ) - it's brain explosion.
and I don't understand why method Set() don't work in my ts :(

### CODE REVIEW

**Comment by jathpr** [comment]
> It's so difficult. I don't fully understand reduce(). on the web are many examples with sum of numbers (it's easy), but when I work with array which is in objects (which are also in an array :) ) - it's brain explosion. and I don't understand why method Set() don't work in my ts :(

Thanks, I will answer in TG chat

**Comment by jathpr** [comment]
and change target branch to your own

**Comment by Yury-Odinson** [comment]
> and change target branch to your own

sorry. stupid mistake

**Comment by jathpr** [comment]
I don't know about Set =(
ask in chat, I remember that someone have the same problem too

do you have "@types/node": "^18.15.11" installed?

**Comment by jathpr** [comment]
> > and change target branch to your own
> 
> sorry. stupid mistake

like all of them (:

**Comment by jathpr** [comment]
and now you have conflicts ⬇️

File: hw_36.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+]
+
+// 1
+const getGenres = (arr: Film[]) => {
```
Comment (jathpr) [inline hw_36.ts:118]: I recommend specific names like `films` against common `arr`

File: hw_36.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    const genresFilms = Array.from(arr).reduce((acc: string[], film) => [...acc, ...film.genre], [])
```
Comment (jathpr) [inline hw_36.ts:119]: arr is already and Array
you don't need `Array.from(arr)` here

File: hw_36.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    return new Set<string>([...genresFilms])
+    // return [...new Set<string>(genresFilms)] <-- This method doesn't working :(
```
Comment (jathpr) [inline hw_36.ts:121]: yes, Set doesn't work with spread
You need `Array.from(setParamHere)`

File: hw_36.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+console.log('--- Task 2 ---')
+console.log(getActors(films))
+
+// 3 
+const getRaiting = (arr: Film[]) => Array.from(arr).sort((a, b) => b.imdbRating - a.imdbRating)
+console.log('--- Task 3 ---')
+console.log(getRaiting(films))
+
+// 4
+const getShortName = (arr: Film[]) => Array.from(arr).map(film => ({id: film.id, title: film.title, released: film.released, plot: film.plot}))
```
Comment (jathpr) [inline hw_36.ts:141]: good
My preferred way
`films.map(({id,title})=>{id,title})`

**Review by jathpr — APPROVED** [review]
fine
but `Array.from(arr)` - is a totally useless construction)

File: hw_36.ts (hunk 2 above)
Comment (Yury-Odinson) [reply hw_36.ts:119]: But without using `Array.from()` I change global array. for example: if I sort the array in the first task without applying `Array.from()` then sorting is applied in subsequent tasks.

File: hw_36.ts (hunk 3 above)
Comment (Yury-Odinson) [reply hw_36.ts:121]: > I don't know about Set =( ask in chat, I remember that someone have the same problem too
> 
> do you have "@types/node": "^18.15.11" installed?

ofcourse. but it wasn't enough. searched in google, but gave up after 30 minutes

File: hw_36.ts (hunk 2 above)
Comment (jathpr) [reply hw_36.ts:119]: yap, for `sort`- you are right. But `map` and `reduce` doesn't mutate the original array. And I think `filter` too

**Pull request merged**